docs/architecture/adr-056-proving-amnesia-attack.md

```diff
@@ -0,0 +1,85 @@
+# ADR 56: Proving Amnesia Attack
+
+## Changelog
+
+- 02.04.20: Initial Draft
+
+## Context
+
+Whilst most created evidence of malicious behaviour is self evident such that any individual can verify them independently there are types of evidence that require collaboration from the network of validators, known as global evidence, in order to accumulate enough information to create evidence that is individually verifiable and can therefore be processed through consensus. Fork Accountability as a whole constitutes of detection, proving and punishing. This ADR addresses how to prove infringement from global evidence.
+
+So far, only [flip flopping](https://github.com/tendermint/spec/blob/master/spec/consensus/light-client/accountability.md#flip-flopping) attacks such as amnesia (which will be what the reactor initially focuses on) and back to the past (only causes light forks) have been identified as forms of global evidence that require more information from other validators.
+
+## Decision
+
+A new reactor will perform the task of turning global evidence to individual evidence. This special circumstance reactor could be combined with the current evidence reactor although the evidence reactor is currently used for gossiping and storing of evidence and so it's purpose would expand. The reactor is initialized when a full node receives PotentialAmnesiaEvidence from a light client.
+
+*REMARK: This means that only the set of full nodes that are directly connected to the light client that observed the misbehaviour will process the evidence. Is this sufficient, do we want to gossip to further nodes, do we want just a single node to process evidence to avoid too much communication overhead?*
+
+```
+type PotentialAmnesiaEvidence struct {
+	C1 types.Commit
+	C2 types.Commit
+}
+```
+
+The receiving full node first runs `Verify()` to ensure that the evidence is valid i.e. both `Commit`s are valid and are of the same height.
```

Comment:
Just so I understand - where in the full "evidence handling process" does this start? Is this the first full node which receives evidence after a light client reports it? Or am I turned around here? It might be helpful to "zoom out" a bit here and describe how this process fits into the bigger picture.

(And is the `PotentialAmnesiaEvidence` above what is being sent from that light client?)

Reply:
https://github.com/tendermint/tendermint/blob/master/docs/architecture/adr-047-handling-evidence-from-light-client.md#decision

Reply:
I should probably add this ^ as a reference as well. I don't know if there is a document we can refer to to give the scope of the entire project - perhaps the fork accountability spec. Alternatively there is an evidence spec but it is empty and should really be written out at some point. I had my eye on doing that sometime in the future after this.

Reply:
Nice change :) 

Fwiw, I'm asking a lot of these questions just because I think this document should answer them. Linking in the discussions can be helpful, but let's make sure they get answered in the ADR itself as well!